src/components/KTreeList/KTreeDraggable.vue

```diff
@@ -0,0 +1,345 @@
+  <VueDraggableNext
+    v-bind="draggableAttrs"
+    :disabled="disableDrag"
+    :list="internalList"
+    :group="{ name: 'k-tree-list', put: !maxLevelReached }"
+    :move="checkMove"
+    :level="level"
+    tag="div"
+    class="k-tree-draggable"
+    @start="onStartDrag"
+    @end="onStopDrag"
+    @change="handleChangeEvent"
+  >
+    <div
+      v-for="element in internalList"
+      :key="element.id"
+      :class="{
+        'has-no-children': hasNoChildren(element)
+      }"
+      class="k-tree-item-container"
+    >
+      <KTreeItem
+        :key="`tree-item-${element.id}-${key}`"
+        :item="element"
+        :disabled="disableDrag"
+        @selected="handleSelectionEvent"
+      >
+        <template #item-icon>
+          <slot
+            :item="element"
+            name="item-icon"
+          >
+            <KIcon
+              v-if="element.icon !== 'none'"
+              :icon="element.icon ? element.icon : 'treeDoc'"
+              :secondary-color="iconSecondaryColor(element)"
+              size="24"
+            />
+          </slot>
+        </template>
+        <template #item-label>
+          <slot
+            :item="element"
+            name="item-label"
+          >
+            {{ element.name }}
+          </slot>
+        </template>
+      </KTreeItem>
+      <KTreeDraggable
+        :key="`tree-item-${element.id}-children-${key}`"
+        :items="getElementChildren(element)"
+        :level="level + 1"
+        :max-level="maxLevel"
+        :disable-drag="disableDrag"
+        :parent-id="element.id"
+        @selected="handleSelectionEvent"
+      >
+        <template #[itemIcon]="slotProps">
+          <slot
+            v-bind="slotProps"
+            name="item-icon"
+          />
+        </template>
+        <template #[itemLabel]="slotProps">
+          <slot
+            v-bind="slotProps"
+            name="item-label"
+          />
+        </template>
+      </KTreeDraggable>
+    </div>
+  </VueDraggableNext>
+</template>
+
+<script lang="ts" setup>
+/**
+ *  Note: if TS errors appear on the
+ *  <template #[itemIcon]="slotProps"> or
+ *  <template #[itemLabel]="slotProps">
+ * lines switch to `v-slot:` notation and Save to let linter clear -->
+ */
+import { computed, ref, watch, onMounted, PropType } from 'vue'
+import { VueDraggableNext } from 'vue-draggable-next'
+import KTreeItem, { TreeListItem } from '@/components/KTreeList/KTreeItem.vue'
+
+export interface ChangeEvent {
+  items: TreeListItem[]
+}
+
+export interface ChildChangeEvent {
+  parent: string,
+  children: TreeListItem[]
+}
+
+export interface DropEvent {
+  item: TreeListItem,
+  parent: TreeListItem
+}
+
+const props = defineProps({
+  items: {
+    type: Array as PropType<TreeListItem[]>,
+    required: true,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  disableDrag: {
+    type: Boolean,
+    default: false,
+  },
+  maxLevel: {
```

Comment:
See suggestion in other file
```suggestion
  maxLevels: {
```